tls/s2n_client_cert.c

```diff
@@ -130,8 +131,15 @@ int s2n_client_cert_recv(struct s2n_connection *conn)
 
     /* Determine the Cert Type, Verify the Cert, and extract the Public Key */
     s2n_pkey_type pkey_type = S2N_PKEY_TYPE_UNKNOWN;
-    POSIX_GUARD_RESULT(s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain_data,
-            cert_chain_size, &pkey_type, &public_key));
+    s2n_result result = s2n_x509_validator_validate_cert_chain(&conn->x509_validator, conn, cert_chain_data,
+            cert_chain_size, &pkey_type, &public_key);
+    if (s2n_result_is_error(result)) {
+        /* reset the stuffer cursor to where cert data start in order to prepare for re-entry, because
+         * s2n_handle_retry_state() will invoke s2n_client_cert_recv() directly without updating stuffer
+         */
```

Comment:
This comment is helpful. Though it might be good to start with saying that `s2n_x509_validator_validate_cert_chain()` handles async callbacks, which may require re-entering `s2n_client_cert_recv()`.